### DISCUSSION TITLE
Trending Ordering in List

### DISCUSSION THREAD

**Discussion by silgon** [pocketbase/pocketbase#5810] [Q&A] [answered]
I've been playing around for a couple of days with pocketbase. I'm wondering about the possibility to make a trending ordering. An analogous sql function would be something like the following.
```sql
   SELECT item_id, like_count, created_at,
          (like_count / EXTRACT(EPOCH FROM (NOW() - created_at))) AS engagement_rate
   FROM likes
   WHERE created_at >= NOW() - INTERVAL '30 days'
   ORDER BY engagement_rate DESC;
```
What would be the way to go? I believe that the standard API would not make it (correct me if mistaken and let me know how) and I have no idea how to give it a go with the pocketbase as a go framework.
Thanks!

**Comment by chnfyi** [comment]
I think you can achieve it using pocketbase as a go framework.

1. Get records using `app.Dao().FindRecordsByExpr` or `app.Dao().FindRecordsByFilter`, filtering relatively to your created_at (now - 30 days) condition.
2. Iterate through this records array to create objects (with item_id, like_count, created_at and engagement_rate properties) that you all append to an new array.
3. Order the new array using `sort.Slice()` with `>` comparison for descending order.

**Reply by silgon** [reply]
It seems to make sense what you are saying. However, I have absolutely no idea how to go about it with the go framework, since I'm completely inexperienced with it (either to create it or to connect it to the client). I'll try to take some time in the following days to implement your idea unless some nice code example comes around before 😅🤞.

**Reply by ganigeorgiev** [reply, marked as answer]
An alternative to custom route is to create a [View collection](https://pocketbase.io/docs/collections/#view-collection) with your custom SQL query but keep in mind that the PocketBase dates are stored in the format `Y-m-d H:i:s.uZ` (e.g. `2024-11-07 08:46:18.792Z`) and you may need to do extra conversion depending on the type of operations (see the builtin [SQLite date functions](https://www.sqlite.org/lang_datefunc.html)).

**Reply by silgon** [reply]
You are a wizard, thank you @ganigeorgiev . Lastly, I had to time to put into this. I created a view collection. I was not actually straight forward since I found the documentation unclear, there was this video (https://www.youtube.com/watch?v=pFfggY55wyM) that helped me. Also, it was weird because I was trying to use sql when I really know it's sqlite 🤦.
Anyway,  I created the field `likes/(julianday('now')-julianday(created))  AS trending` in the query (for those interested)
